common/flatpak-context.c

```diff
@@ -894,6 +894,111 @@ flatpak_context_add_nousb_query (FlatpakContext        *context,
   flatpak_context_add_query_to (context->hidden_usb_devices, usb_query);
 }
 
+static gboolean
+flatpak_context_add_usb_list (FlatpakContext *context,
+                              const char     *list,
+                              GError        **error)
+{
+  return flatpak_context_parse_usb_list (list, context->enumerable_usb_devices,
+					 context->hidden_usb_devices, error);
+}
+
+static gboolean
+flatpak_context_add_usb_list_from_file (FlatpakContext *context,
+                                        const char     *path,
+                                        GError        **error)
+{
+  g_autofree char *contents = NULL;
+
+  if (!flatpak_validate_path_characters (path, error))
+    return FALSE;
+
+  if (!g_file_get_contents (path, &contents, NULL, error))
+    return FALSE;
+
+  return flatpak_context_parse_usb_list (contents, context->enumerable_usb_devices,
+					 context->hidden_usb_devices, error);
+}
+
+gboolean
+flatpak_context_parse_usb_list (const char     *buffer,
+                                GHashTable     *enumerable,
+                                GHashTable     *hidden,
+                                GError        **error)
+{
+  char *aux = NULL;
+  g_autoptr(FlatpakUsbQuery) usb_query = NULL;
+  g_autoptr(GInputStream) stream = NULL;
+  g_autoptr(GDataInputStream) buffered = NULL;
+  g_autoptr(GError) local_error = NULL;
+
+  stream = g_memory_input_stream_new_from_data (buffer, -1, NULL);
+  if (!stream)
+    return FALSE;
+
+  buffered = g_data_input_stream_new (G_INPUT_STREAM (stream));
+
+  while ((aux = g_data_input_stream_read_line (buffered, NULL, NULL, &local_error)))
+    {
+      g_autofree char *line = g_steal_pointer (&aux);
+      g_auto(GStrv) split = NULL;
+      GStrv current = NULL;
+      gboolean blocking = FALSE;
+
+      if (line[0] == '#')
+        {
+          continue;
+        }
+
+      split = g_strsplit (line, ";", 0);
+      if (!split || !*split)
+        {
+          continue;
+        }
+
+      current = split;
+      while (*current)
```

Comment:
I'll be honest, I've grown quite a bit of a distaste to iterating `GStrv`s like using this fragile mechanics of pointer manipulation. Please let me suggest this instead:

```suggestion
      for (size_t i = 0; split != NULL && spit[i] != NULL; i++)
```

Reply:
note that this being C, using array subscript brings no change in safety whatsoever.

Reply:
CodeQL figured out that `split` can never be `NULL` here and this could be

```C
      for (size_t i = 0; spit[i] != NULL; i++)
```